service/topology_coordinator.cc

```diff
@@ -1665,6 +1670,14 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
             co_return;
         }
 
+        // generate updates for resize finalizations
+        if (!plan.resize_plan().finalize_resize.empty() && !utils::get_local_injector().enter("tablet_split_finalization_postpone")) {
+            // Executes a global barrier to guarantee that any process (e.g. repair) holding stale version
+            // of token metadata will complete before we update topology.
+            guard = co_await global_tablet_token_metadata_barrier(std::move(guard));
+            co_await generate_resize_finalization_updates(updates, guard, get_token_metadata_ptr(), plan.resize_plan().finalize_resize);
+        }
```

Comment:
It's better to not expose resize finalization details here. It's also missing that the resize decision might have changed and we don't have to finalize it anymore. see handle_tablet_resize_finalization() recalls balance_tablets() after executing barrier.

I suggest we extract everything from handle_tablet_resize_finalization() (with exception of update topology part) into generate_resize_finalization_updates() which can be called here unconditionally.

Reply:
I have updated the code.